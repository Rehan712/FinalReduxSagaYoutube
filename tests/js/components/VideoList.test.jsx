import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { VideoList } from '../../../src/js/components/VideoList';

Enzyme.configure({ adapter: new Adapter() });

describe('Test VideoList Component', () => {
	let videoList;
	beforeEach(() => {
		videoList = [
			{
				id: { videoId: '' },
				snippet: {
					title: 'naruto',
					description: 'ninja',
					thumbnails: {
						default: {
							url: 'naruto.com'
						}
					}
				}
			},
			{
				id: { videoId: '' },
				snippet: {
					title: 'naruto uzumaki',
					description: 'ninja clan',
					thumbnails: {
						default: {
							url: 'narutouzamaki.com'
						}
					}
				}
			}
		];
	});

	const error = {
		message: 'Api Failure'
	};
	const selectVideo = jest.fn();

	it('test the rendering of the component', () => {
		const wrapper = shallow(
			<VideoList
				selectVideo={selectVideo}
				isLoading={true}
				error={null}
				videoList={videoList}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});
	it('test the rendering of the component', () => {
		const wrapper = shallow(
			<VideoList
				selectVideo={selectVideo}
				isLoading={false}
				error={null}
				videoList={videoList}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('test the rendering of the component', () => {
		const wrapper = shallow(
			<VideoList
				selectVideo={selectVideo}
				isLoading={false}
				error={error}
				videoList={videoList}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});
	it('passes on the selectVideo fn', () => {
		expect(selectVideo.mock.calls.length).toBe(0);
	});
});
