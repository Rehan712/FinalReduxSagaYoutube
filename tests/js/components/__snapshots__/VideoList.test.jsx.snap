// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test VideoList Component test the rendering of the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoList
    error={null}
    isLoading={true}
    selectVideo={[MockFunction]}
    videoList={
      Array [
        Object {
          "id": Object {
            "videoId": "",
          },
          "snippet": Object {
            "description": "ninja",
            "thumbnails": Object {
              "default": Object {
                "url": "naruto.com",
              },
            },
            "title": "naruto",
          },
        },
        Object {
          "id": Object {
            "videoId": "",
          },
          "snippet": Object {
            "description": "ninja clan",
            "thumbnails": Object {
              "default": Object {
                "url": "narutouzamaki.com",
              },
            },
            "title": "naruto uzumaki",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        Loading Videos.....
      </div>,
      "className": "videoList",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading Videos.....",
      },
      "ref": null,
      "rendered": "Loading Videos.....",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          Loading Videos.....
        </div>,
        "className": "videoList",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading Videos.....",
        },
        "ref": null,
        "rendered": "Loading Videos.....",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test VideoList Component test the rendering of the component 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoList
    error={null}
    isLoading={false}
    selectVideo={[MockFunction]}
    videoList={
      Array [
        Object {
          "id": Object {
            "videoId": "",
          },
          "snippet": Object {
            "description": "ninja",
            "thumbnails": Object {
              "default": Object {
                "url": "naruto.com",
              },
            },
            "title": "naruto",
          },
        },
        Object {
          "id": Object {
            "videoId": "",
          },
          "snippet": Object {
            "description": "ninja clan",
            "thumbnails": Object {
              "default": Object {
                "url": "narutouzamaki.com",
              },
            },
            "title": "naruto uzumaki",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <VideoListEntry
          handleSelectedVideo={[MockFunction]}
          video={
            Object {
              "id": Object {
                "videoId": "",
              },
              "snippet": Object {
                "description": "ninja",
                "thumbnails": Object {
                  "default": Object {
                    "url": "naruto.com",
                  },
                },
                "title": "naruto",
              },
            }
          }
        />,
        <VideoListEntry
          handleSelectedVideo={[MockFunction]}
          video={
            Object {
              "id": Object {
                "videoId": "",
              },
              "snippet": Object {
                "description": "ninja clan",
                "thumbnails": Object {
                  "default": Object {
                    "url": "narutouzamaki.com",
                  },
                },
                "title": "naruto uzumaki",
              },
            }
          }
        />,
      ],
      "className": "videoList",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleSelectedVideo": [MockFunction],
          "video": Object {
            "id": Object {
              "videoId": "",
            },
            "snippet": Object {
              "description": "ninja",
              "thumbnails": Object {
                "default": Object {
                  "url": "naruto.com",
                },
              },
              "title": "naruto",
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleSelectedVideo": [MockFunction],
          "video": Object {
            "id": Object {
              "videoId": "",
            },
            "snippet": Object {
              "description": "ninja clan",
              "thumbnails": Object {
                "default": Object {
                  "url": "narutouzamaki.com",
                },
              },
              "title": "naruto uzumaki",
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <VideoListEntry
            handleSelectedVideo={[MockFunction]}
            video={
              Object {
                "id": Object {
                  "videoId": "",
                },
                "snippet": Object {
                  "description": "ninja",
                  "thumbnails": Object {
                    "default": Object {
                      "url": "naruto.com",
                    },
                  },
                  "title": "naruto",
                },
              }
            }
          />,
          <VideoListEntry
            handleSelectedVideo={[MockFunction]}
            video={
              Object {
                "id": Object {
                  "videoId": "",
                },
                "snippet": Object {
                  "description": "ninja clan",
                  "thumbnails": Object {
                    "default": Object {
                      "url": "narutouzamaki.com",
                    },
                  },
                  "title": "naruto uzumaki",
                },
              }
            }
          />,
        ],
        "className": "videoList",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleSelectedVideo": [MockFunction],
            "video": Object {
              "id": Object {
                "videoId": "",
              },
              "snippet": Object {
                "description": "ninja",
                "thumbnails": Object {
                  "default": Object {
                    "url": "naruto.com",
                  },
                },
                "title": "naruto",
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleSelectedVideo": [MockFunction],
            "video": Object {
              "id": Object {
                "videoId": "",
              },
              "snippet": Object {
                "description": "ninja clan",
                "thumbnails": Object {
                  "default": Object {
                    "url": "narutouzamaki.com",
                  },
                },
                "title": "naruto uzumaki",
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test VideoList Component test the rendering of the component 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoList
    error={
      Object {
        "message": "Api Failure",
      }
    }
    isLoading={false}
    selectVideo={[MockFunction]}
    videoList={
      Array [
        Object {
          "id": Object {
            "videoId": "",
          },
          "snippet": Object {
            "description": "ninja",
            "thumbnails": Object {
              "default": Object {
                "url": "naruto.com",
              },
            },
            "title": "naruto",
          },
        },
        Object {
          "id": Object {
            "videoId": "",
          },
          "snippet": Object {
            "description": "ninja clan",
            "thumbnails": Object {
              "default": Object {
                "url": "narutouzamaki.com",
              },
            },
            "title": "naruto uzumaki",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        Api Failure
      </div>,
      "className": "videoList",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Api Failure",
      },
      "ref": null,
      "rendered": "Api Failure",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          Api Failure
        </div>,
        "className": "videoList",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Api Failure",
        },
        "ref": null,
        "rendered": "Api Failure",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
